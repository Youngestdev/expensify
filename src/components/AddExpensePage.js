import React from "react";

const AddExpensePage = () => <div>Add some expenses here</div>;

export default AddExpensePage;
