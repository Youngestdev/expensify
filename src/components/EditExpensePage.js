import React from "react";
import { connect } from 'react-redux'

const EditExpensePage = props => <div>Editing the expense with an id of {props.match.params.id}</div>;
;

export default EditExpensePage;
