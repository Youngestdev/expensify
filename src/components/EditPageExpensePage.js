import React from "react";

const EditPageExpensePage = props => {
  console.log(props);
  return <div>Editing the expense with an id of {props.match.params.id}</div>;
};

export default EditPageExpensePage;
