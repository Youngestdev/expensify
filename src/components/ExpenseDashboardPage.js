import React from "react";

const ExpenseDashboardPage = () => <div>This is coming from dashboard component</div>;

export default ExpenseDashboardPage;
