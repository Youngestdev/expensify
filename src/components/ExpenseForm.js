import React, { Component } from 'react';

export default class ExpenseForm extends Component {
  render() {
    return (
      <div>
        <h2>Expense Form</h2>
      </div>
    )
  }
}