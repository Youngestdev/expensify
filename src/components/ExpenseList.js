import React from "react";
import { connect } from "react-redux";

const ExpenseList = props => (
  <div>
    <h2>Expense List</h2>
    {props.expenses.length}
  </div>
);

const ConnectedExpenseList = connect(state => {
  return {
    expenses: state.expenses
  };
})(ExpenseList);

export default ConnectedExpenseList;
