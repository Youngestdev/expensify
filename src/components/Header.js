import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <div>
    <h2>Expensify</h2>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
  </div>
);

export default Header;
