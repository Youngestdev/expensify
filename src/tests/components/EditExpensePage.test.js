import React from 'react'
import { shallow } from 'enzyme'
import Expenses from '../fixtures/expenses'
import { EditExpensePage } from '../../components/EditExpensePage'

let editExpense, removeExpense, history, wrapper

beforeEach(() => {
  editExpense = jest.fn()
  removeExpense = jest.fn()
  history = { push: jest.fn() }
  wrapper = shallow(<EditExpensePage
    editExpense={editExpense}
    removeExpense={removeExpense}
    history={history}
    expense={Expenses[2]}
  />)
})

test('should render EditExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle edit expense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(Expenses[2]);
  expect(history.push).toHaveBeenLastCalledWith('/')
  expect(editExpense).toHaveBeenLastCalledWith(Expenses[2].id, Expenses[2])
})

test('should handle remove expense', () => {
  wrapper.find('button').simulate('click')
  expect(history.push).toHaveBeenLastCalledWith('/')
  expect(removeExpense).toHaveBeenLastCalledWith({
    id: Expenses[2].id
  })
})