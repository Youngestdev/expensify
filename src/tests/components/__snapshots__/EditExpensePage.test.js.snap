// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 943,
        "createdAt": 345600000,
        "description": "Domain",
        "id": 3,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    remove
  </button>
</div>
`;
